src/lib/frontend/d_cnf.ml

```diff
@@ -1565,13 +1565,20 @@ let rec mk_expr
           in
           let hyp = List.map aux_mk_expr hyp in
 
-          (* triggers *)
-          let trgs =
+          (* triggers - on quantifier for AE, on body for SMT-LIB *)
+          let root_trgs =
             begin match DStd.Tag.get root_tags DE.Tags.triggers with
               | Some t -> t
               | _ -> []
             end
           in
+          let body_trgs =
+            begin match DStd.Tag.get body.term_tags DE.Tags.triggers with
+              | Some t -> t
+              | _ -> []
+            end
+          in
+          let trgs = root_trgs @ body_trgs in
```

Comment:
To my understanding, `root_trgs` or `body_trgs` are disjoint? I understand you use the concatenation operator here because `l @ []` and `[] @ l` are reduced immediately to `l`, but for sake of readability, I would prefer to match on the logic stored in the state of the typer as you did here:
```ocaml
let builtins =
  fun _st (lang : Typer.lang) ->
  match lang with
  | `Logic Alt_ergo -> ae_fpa_builtins
  | `Logic (Smtlib2 _) -> bv_builtins ++ smt_fpa_builtins
  | _ -> fun _ _ -> `Not_found
```

Reply:
I was going to say that doing a concatenation this way is more robust to potential future Dolmen changes, but actually the code as written is buggy in the case of nested quantifiers in SMT (we pick up the pattern for both the outer and inner quantifier), good catch!